src/modules/universal-discovery/components/finder/finder.component.js

```diff
@@ -300,6 +301,8 @@ export default class FinderComponent extends Component {
 
         this.setState(() => ({ lastSelectedItem: parent }));
 
+        this.lastSelectedLocationId = location.id;
```

Comment:
I don't like the name. I will be confusing.
We have: 
`lastSelectedItem` - and this is parent
`lastSelectedLocationId` and this is the id of current selection?